bitcoin/src/network/constants.rs

```diff
@@ -144,7 +166,7 @@ impl FromStr for Network {
             "testnet" => Testnet,
             "signet" => Signet,
             "regtest" => Regtest,
-            _ => return Err(ParseNetworkError(s.to_owned())),
+            _ => return Network::from_core_arg(s),
```

Comment:
I'm not really convinced this is a good idea. It's basically [robustness principle which is harmful](https://www.ietf.org/archive/id/draft-iab-protocol-maintenance-05.html)

(Interestingly the frequency of attempts at robustness principle I replied to is unusually high last few months...)

Reply:
However we decide to do it, accepting "mainnet" seems like a win to me, bitcoins overloading of "bitcoin" for the network is annoying when we have a term used in the common vernacular that better describes it.

Reply:
Yeah, I'm open to that one. I'm even tempted to drop `bitcoin` but I think it was already changed in the past and this would break user-facing code if blindly upgraded. E.g. electrs uses this in argument/configuration parsing.

Reply:
This is mainly why I went for keeping both instead of dropping `bitcoin`. I'm happy with any of the options: either accepting what is used in common vernacular (main/mainnet, test/testnet) or going for only what Core returns (main/test).